src/Symfony/Bundle/FrameworkBundle/Tests/Command/RouterDebugCommandTest.php

```diff
@@ -0,0 +1,186 @@
+        $this->assertStringNotContainsString('! [CAUTION] The routes list is not sorted in the parsing order.', $tester->getDisplay(true));
+    }
+
+    public function testSortingByName()
+    {
+        $tester = $this->createCommandTester();
+        $result = $tester->execute(['--sort' => 'name'], ['decorated' => false]);
+        $this->assertSame(0, $result, 'Returns 0 in case of success');
+        $this->assertRegExp('/(alfa).*\n\W*(bravo).*\n\W*(charlie).*\n\W*(delta)/m', $tester->getDisplay(true));
+        $this->assertStringContainsString('! [CAUTION] The routes list is not sorted in the parsing order.', $tester->getDisplay(true));
+    }
+
+    public function testSortingByPath()
+    {
+        $tester = $this->createCommandTester();
+        $result = $tester->execute(['--sort' => 'path'], ['decorated' => false]);
+        $this->assertSame(0, $result, 'Returns 0 in case of success');
+        $this->assertRegExp('/(\/romeo).*\n.*(\/sierra).*\n.*(\/tango).*\n.*(\/uniform)/m', $tester->getDisplay(true));
+        $this->assertStringContainsString('! [CAUTION] The routes list is not sorted in the parsing order.', $tester->getDisplay(true));
+    }
+
+    public function testThrowsExceptionWithInvalidParameter()
+    {
+        $tester = $this->createCommandTester();
+        $this->expectExceptionMessage('The option "foobar" is not valid');
+        $tester->execute(['--sort' => 'foobar'], ['decorated' => false]);
+    }
+
+    public function testThrowsExceptionWithNullParameter()
+    {
+        $tester = $this->createCommandTester();
+        $this->expectExceptionMessage('The "--sort" option requires a value.');
+        $tester->execute(['--sort' => null], ['decorated' => false]);
+    }
+
+    public function testSortingByPriorityWithDuplicatePath()
+    {
+        $tester = $this->createCommandTesterWithDuplicatePath();
+        $result = $tester->execute(['--sort' => 'priority'], ['decorated' => false]);
+        $this->assertSame(0, $result, 'Returns 0 in case of success');
+        $this->assertRegExp('/(charlie).*\n\W*(alfa).*\n\W*(delta).*\n\W*(bravo).*\n\W*(echo)/m', $tester->getDisplay(true));
+        $this->assertStringNotContainsString('! [CAUTION] The routes list is not sorted in the parsing order.', $tester->getDisplay(true));
+    }
+
+    public function testSortingByNameWithDuplicatePath()
+    {
+        $tester = $this->createCommandTesterWithDuplicatePath();
+        $result = $tester->execute(['--sort' => 'name'], ['decorated' => false]);
+        $this->assertSame(0, $result, 'Returns 0 in case of success');
+        $this->assertRegExp('/(alfa).*\n\W*(bravo).*\n\W*(charlie).*\n\W*(delta).*\n\W*(echo)/m', $tester->getDisplay(true));
+        $this->assertStringContainsString('! [CAUTION] The routes list is not sorted in the parsing order.', $tester->getDisplay(true));
+    }
+
+    public function testSortingByPathWithDuplicatePath()
```

Comment:
this test should actually also make sure the routes with the same path stay in the correct order as before (check routes names) as I mention in the stable sorting comment